crates/ty_python_semantic/src/types/protocol_class.rs

```diff
@@ -403,10 +408,158 @@ impl<'db> ProtocolMemberData<'db> {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, salsa::Update, get_size2::GetSize)]
+struct PropertyMember<'db> {
+    get_type: Option<Type<'db>>,
+    set_type: Option<Type<'db>>,
+}
+
+impl<'db> PropertyMember<'db> {
+    fn from_property_instance(property: PropertyInstanceType<'db>, db: &'db dyn Db) -> Self {
+        PropertyMember {
+            // if the getter or setter is a function literal, we should
+            // upcast it to a `CallableType` so that two protocols with identical property
+            // members are recognized as equivalent.
+            get_type: property
+                .getter(db)
+                .map(|ty| ty.into_callable(db).unwrap_or(ty)),
+            set_type: property
+                .setter(db)
+                .map(|ty| ty.into_callable(db).unwrap_or(ty)),
+        }
+    }
```

Comment:
Thank you very much for this PR -- it was really helpful! This still doesn't go _quite_ in the direction I was thinking of, however -- you can see https://github.com/astral-sh/ruff/pull/19936 for the kind of thing I had in mind.

I think what I'd like to do at this point is to first land https://github.com/astral-sh/ruff/pull/20095 -- which includes many of your test improvements, plus some of my own -- and then land a cleaned-up version of #19936 on top of that. I'll list you as a co-author of both PRs, since both build on work you've contributed. I hope that's okay!

Reply:
OK, let's move on!